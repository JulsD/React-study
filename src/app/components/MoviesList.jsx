import React from 'react';
import MovieItem from './MovieItem';

class MoviesList extends React.Component {

  constructor(props) {
    super(props);

  }

  render() {
    const movies = this.props.movies;
    const listMovies = movies.map((movie) =>
      <MovieItem key={movie.toString()} title={movie} />
    );
    return (
      <div className="movies-wrapper">
        {listMovies}
      </div>
    );
  }

}

export default MoviesList;
