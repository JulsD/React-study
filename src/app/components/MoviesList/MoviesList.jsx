import React from 'react';

import { MovieItem } from './../';
import styles from './movies-list.css';

class MoviesList extends React.Component {

  constructor(props) {
    super(props);

  }

  render() {
    const movies = this.props.movies;
    const listMovies = movies.map((movie) =>
      <MovieItem key={movie.toString()} title={movie} />
    );
    return (
      <div className = {styles.root}>
        {listMovies}
      </div>
    );
  }

}

export default MoviesList;
