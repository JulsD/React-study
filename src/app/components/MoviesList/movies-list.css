.root {
  display: flex;
  flex-wrap: wrap;
}
.root > * {
  flex: 0 1 33%;
  box-sizing: border-box;
}