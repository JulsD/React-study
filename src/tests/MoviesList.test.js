import React from 'react';
import MoviesList from '../app/components/MoviesList';

test('MoviesList should render a <Link>', () => {
  const movies = [{ id: 1 }, { id: 2 },{ id: 3 }];
  const wrapper = shallow(
    <MoviesList movies={movies}/>
  );
  expect(wrapper).toMatchSnapshot();
});
