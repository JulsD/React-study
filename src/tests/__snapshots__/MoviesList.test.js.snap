// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should render a <Link> 1`] = `
<div
  className="root"
>
  <MovieItem
    key="1"
    movie={
      Object {
        "id": 1,
      }
    }
  />
  <MovieItem
    key="2"
    movie={
      Object {
        "id": 2,
      }
    }
  />
  <MovieItem
    key="3"
    movie={
      Object {
        "id": 3,
      }
    }
  />
</div>
`;
